### ISSUE TITLE
Enable missing_value to override default default_value for fields

### ISSUE THREAD

**Issue by mgduda** [MPAS-Dev/MPAS-Model#551]
Following a discussion on 4 May 2020, there is interest in having the the default value for a field take on the specified `missing_value` if no explicit `default_value` is given in the registry entry for a field.

The motivation for this change is that fields that do contain missing values would otherwise need to be explicitly initialized by a core to the missing value; initializing a field to its registry-specified `missing_value` would remove this burden from the core developer.

**Comment by xylar** [comment]
@mgduda, yep, that looks like what we discussed.  Thanks for being willing to implement this!

**Comment by MiCurry** [comment]
And if both `missing_value` and `default_value` are present, then the field's default value should be set to `default_value`? i.e. the current behavior?

**Comment by xylar** [comment]
@MiCurry, my hope is that `missing_value` and `default_value` wouldn't be specified for the same field because that would be confusing.

@mgduda, is the difference in behavior between `missing_value` and `default_value` whether this attribute ends up in the NetCDF file or not (`missing_value`: yes; `default_value`: no)?

**Comment by mgduda** [comment]
@xylar Aside from differences in how these two attributes are used to set the initial value of a field after it has been allocated, that's correct: the `missing_value` attribute is added as a netCDF variable attribute, while the `default_value` attribute is not.
